nshlib/nsh_envcmds.c

```diff
@@ -533,29 +533,45 @@ int cmd_unset(FAR struct nsh_vtbl_s *vtbl, int argc, char **argv)
 {
 #if defined(CONFIG_NSH_VARS) || !defined(CONFIG_DISABLE_ENVIRON)
   int status;
+  FAR char *oldvalue;
```

Comment:
this is inside wrong ifdefs. should be:
#if defined(CONFIG_NSH_VARS) && !defined(CONFIG_DISABLE_ENVIRON)

will fix this